File: testdata/io/analyse.ts
import fs from 'node:fs';
import axios from 'axios';

export function chargeCustomer() {
  fs.readFileSync('in.txt', 'utf8');
  fetch('https://example.com/items');
  axios.post('/charge', {});
}

const loadLedger = () => {
  fs.promises.readFile('ledger.txt', 'utf8');
  axios.get('/ledger');
};

const saveLedger = () => {
  fs.writeFileSync('ledger.txt', 'x');
  process.stdout.write('ok');
};

class PaymentService {
  Charge() {
    fs.readFileSync('a.txt', 'utf8');
    fs.writeFileSync('b.txt', 'value');
    axios.patch('/payment', {});
  }
}

void loadLedger;
void saveLedger;
void PaymentService;

File: testdata/dart/io/analyse.dart
import 'dart:io';

Future<void> loadData() async {
  await File('input.txt').readAsString();
  await http.get(Uri.parse('https://example.com/items'));
  await dio.get('/items');
}

void saveData() {
  File('output.txt').writeAsString('ok');
  stdout.writeln('saved');
  print('done');
  http.post(Uri.parse('https://example.com/items'));
  dio.patch('/items');
}

void idle() {}

class ApiClient {
  Future<void> fetchData() async {
    await File(path).openRead();
    await client.get(Uri.parse('https://example.com/items'));
  }

  Future<void> pushData() async {
    await File(path).writeAsBytes(bytes);
    stdout.write('x');
    await client.post(Uri.parse('https://example.com/items'));
    http.put(Uri.parse('https://example.com/items'));
    await dio.post('/items');
  }

  void noop() {}
}
